src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -303,9 +303,14 @@ public Future<Void> updateOrder(CompositePurchaseOrder compPO, boolean deleteHol
               }
               compPO.getCompositePoLines().forEach(poLine -> PoLineCommonUtil.updateLocationsQuantity(poLine.getLocations()));
               return openCompositeOrderFlowValidator.checkLocationsAndPiecesConsistency(compPO.getCompositePoLines(), requestContext);
-            } else {
-              return Future.succeededFuture();
             }
+            return Future.succeededFuture();
+          })
+          .compose(v -> {
```

Comment:
We have 1 compose per each state transition, so lets keep the initial structure, you can just this validation check by composing checkLocationsAndPiecesConsistency

Reply:
done